python/ray/tune/trial.py

```diff
@@ -288,6 +291,17 @@ def __init__(self,
         self.last_update_time = -float("inf")
         self.checkpoint_freq = checkpoint_freq
         self.checkpoint_at_end = checkpoint_at_end
+        self.keep_checkpoints_num = keep_checkpoints_num
+        self.history = []
+        self.checkpoint_score_attr = checkpoint_score_attr
+        self._cmp_greater = True
+        self.best_checkpoint_attr_value = -float("inf")
+        self.results_since_checkpoint_sum = 0
+        self.results_since_checkpoint_cnt = 0
+        if checkpoint_score_attr and checkpoint_score_attr.startswith("min-"):
+            self._cmp_greater = False
+            self.checkpoint_score_attr = checkpoint_score_attr[4:]
+            self.best_checkpoint_attr_value = float("inf")
```

Comment:
Stylistically, this is a lot of class attributes to add for one feature. Could it be done with fewer (how about: `self.keep_checkpoints_num`, `self.checkpoint_score_attr`, `self.best_checkpoint_history`)?

Reply:
I agree and have changed it a bit, although the number of attributes is the same.